de.bund.bfr.knime.fsklab.service/src/de/bund/bfr/knime/fsklab/service/FskService.java

```diff
@@ -78,275 +81,277 @@
-      throws IOException, URISyntaxException {
-    URL fileURL = bundle.getEntry(path);
-    URL resolvedFileURL = FileLocator.toFileURL(fileURL);
-    URI resolvedURI = new URI(resolvedFileURL.getProtocol(), resolvedFileURL.getPath(), null);
-    return new File(resolvedURI);
-  }
+	private static final String MIME_JSON = "application/json";
+
+	private static final NodeLogger LOGGER = NodeLogger.getLogger(FskService.class);
+
+	private static final JsonTransformer jsonTransformer = new JsonTransformer();
+
+	private final ConversionUtils utils = new ConversionUtils();
+
+	private int port;
+
+	public int getPort() {
+		return port;
+	}
+
+	@Override
+	public void run() {
+
+		try {
+			initDatabase();
+		} catch (SQLException | ClassNotFoundException e1) {
+			LOGGER.error("Initializing DB", e1);
+			return;
+		}
+
+		port(0);
+
+		// Enable CORS
+		options("/*", (request, response) -> {
+
+			String accessControlRequestHeaders = request.headers("Access-Control-Request-Headers");
+			if (accessControlRequestHeaders != null) {
+				response.header("Access-Control-Allow-Headers", accessControlRequestHeaders);
+			}
+
+			String accessControlRequestMethod = request.headers("Access-Control-Request-Method");
+			if (accessControlRequestMethod != null) {
+				response.header("Access-Control-Allow-Methods", accessControlRequestMethod);
+			}
+
+			return "OK";
+		});
+
+		before((request, response) -> response.header("Access-Control-Allow-Origin", "*"));
+
+		get("getById/:vocabulary/:id", (req, res) -> {
+			try (Connection connection = DriverManager.getConnection("jdbc:h2:~/.fsk/vocabularies")) {
+				res.type(MIME_JSON);
+				BasicRepository<?> repository = getRepository(req.params(":vocabulary"), connection);
+				int id = Integer.parseInt(req.params(":id"));
+				return repository.getById(id);
+			}
+		}, jsonTransformer);
+
+		get("/getAll/:vocabulary", (req, res) -> {
+			try (Connection connection = DriverManager.getConnection("jdbc:h2:~/.fsk/vocabularies")) {
+				res.type(MIME_JSON);
+				BasicRepository<?> repository = getRepository(req.params(":vocabulary"), connection);
+				return repository.getAll();
+			}
+		}, jsonTransformer);
+
+		get("/getAllNames/:vocabulary", (req, res) -> {
+			try (Connection connection = DriverManager.getConnection("jdbc:h2:~/.fsk/vocabularies")) {
+				res.type(MIME_JSON);
+				BasicRepository<?> repository = getRepository(req.params(":vocabulary"), connection);
+				return repository.getAllNames();
+			}
+		}, jsonTransformer);
+
+		// input metadata as body parameter
+		post("convertMetadata/:targetModelClass", (req, res) -> {
+			res.type("application/json");
+
+			try {
+				JsonNode inputMetadata = jsonTransformer.MAPPER.readTree(req.body());
+				String targetClass = req.params("targetModelClass");
+				JsonNode convertedMetadata = utils.convertModel(inputMetadata, targetClass);
+
+				res.status(200);
+				return convertedMetadata;
+			} catch (Exception err) {
+				res.status(400);

```

Comment:
This could be improved in the future since 400 is not describing always  good what really happened.
I would prefer to use 500 error code rather than using 400.
the 400 can be used to notify that the endpoint received not well formatted request parameters.
But for now It's fine and the algorithm works for my real testing.

Reply:
Thanks! I will open an issue for this.